frontend/wallet/src/render/contexts/SettingsProvider.re

```diff
@@ -1,8 +1,33 @@
+open Tc;
+
 module SettingsContextType = {
-  type t = (option(Settings.t), Settings.t => unit);
+  type t = (
+    option(Settings.t),
+    (option(Settings.t) => option(Settings.t)) => unit,
+  );
 
   let initialContext = (None, _ => ());
 };
 
 type t = SettingsContextType.t;
 include ContextProvider.Make(SettingsContextType);
+
+let createContext = () => {
+  let (settings, setSettings) =
+    React.useState(() => Result.toOption(SettingsRenderer.loadSettings()));
+
+  (
+    settings,
+    createNewSettings =>
+      setSettings(settings =>
+        createNewSettings(settings)
+        |> Option.map(~f=newSettings => {
+             Task.attempt(
+               ~f=_ => (),
+               SettingsRenderer.saveSettings(newSettings),
```

Comment:
This may race the writes, we can fix later by writing to a temp file and doing an atomic copy. See #2473